backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -1062,4 +1111,149 @@ private String getStringFromLocationList(List<String> locationList) {
     }
     return blobContainerList;
   }
+
+  private Map<String, Object> getRootShaListByCommitsOrRepos(
+      Session session, FindRepositoriesBlobs request) {
+
+    Map<String, Object> parametersMap = new HashMap<>();
+
+    String alias = "cm";
+    StringBuilder rootQueryStringBuilder =
+        new StringBuilder(" FROM ")
+            .append(CommitEntity.class.getSimpleName())
+            .append(" ")
+            .append(alias)
+            .append(" ");
+
+    StringBuilder joinClause = new StringBuilder();
+    joinClause
+        .append(" INNER JOIN ")
+        .append(InternalFolderElementEntity.class.getSimpleName())
+        .append(" folderElm ")
+        .append(" ON ");
+    joinClause.append("folderElm.folder_hash = ").append(alias).append(".rootSha ");
+
+    StringBuilder whereClause = new StringBuilder();
+    boolean appendAND = false;
+    if (!request.getRepoIdsList().isEmpty()) {
```

Comment:
use the AND joiner logic which you did on a different PR

Reply:
Done